test/e2e/chaos/stresschaos/cpu.go

```diff
@@ -0,0 +1,130 @@
+
+package stresschaos
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestcaseMemoryStressInjectionOnceThenRecover(
+	ns string,
+	cli client.Client,
+	peers []*corev1.Pod,
+	ports []uint16,
+	c http.Client,
+) {
+	ctx := context.Background()
+	By("create memory stress chaos CRD objects")
+	memoryStressChaos := makeMemoryStressChaos(ns, "memory-stress", ns, "stress-peer-0", "50M", 1)
+	err := cli.Create(ctx, memoryStressChaos.DeepCopy())
+	framework.ExpectNoError(err, "create stresschaos error")
+
+	By("waiting for assertion some pods are experiencing memory stress ")
+	err = wait.Poll(time.Second, 15*time.Second, func() (done bool, err error) {
+		conditions, err := probeStressCondition(c, peers, ports)
+		if err != nil {
+			return false, err
+		}
+		if conditions[0].MemoryUsage > 50*1024*1024 && conditions[1].MemoryUsage < 50*1024*1024 {
+			return true, nil
+		}
+		framework.Logf("get Memory: [%d, %d]", conditions[0].MemoryUsage, conditions[1].MemoryUsage)
+		return false, nil
+	})
+	framework.ExpectNoError(err, "memory stress failed")
+	By("delete pod failure chaos CRD objects")
+
+	err = cli.Delete(ctx, memoryStressChaos.DeepCopy())
+	framework.ExpectNoError(err, "delete stresschaos error")
+	By("waiting for assertion recovering")
+	err = wait.Poll(time.Second, 15*time.Second, func() (done bool, err error) {
+		conditions, err := probeStressCondition(c, peers, ports)
+		if err != nil {
+			return false, err
+		}
+		if conditions[0].MemoryUsage < 50*1024*1024 && conditions[1].MemoryUsage < 50*1024*1024 {
+			return true, nil
+		}
+		framework.Logf("get Memory: [%d, %d]", conditions[0].MemoryUsage, conditions[1].MemoryUsage)
+		return false, nil
+	})
+	framework.ExpectNoError(err, "fail to recover from memory stress")
+}
+
+func TestcaseCPUStressInjectionOnceThenRecover(
+	ns string,
+	cli client.Client,
+	peers []*corev1.Pod,
+	ports []uint16,
+	c http.Client,
+) {
+	ctx := context.Background()
+	By("create cpu stress chaos CRD objects")
+	cpuStressChaos := makeCPUStressChaos(ns, "cpu-stress", ns, "stress-peer-0", 1, 100)
+	err := cli.Create(ctx, cpuStressChaos.DeepCopy())
+	framework.ExpectNoError(err, "create stresschaos error")
+
+	lastCPUTime := make([]uint64, 2)
+	diff := make([]uint64, 2)
+	By("waiting for assertion some pods are experiencing cpu stress ")
+	err = wait.Poll(time.Second, 15*time.Second, func() (done bool, err error) {
+		conditions, err := probeStressCondition(c, peers, ports)
+		if err != nil {
+			return false, err
+		}
+
+		diff[0] = conditions[0].CpuTime - lastCPUTime[0]
+		diff[1] = conditions[1].CpuTime - lastCPUTime[1]
+		lastCPUTime[0] = conditions[0].CpuTime
+		lastCPUTime[1] = conditions[1].CpuTime
+		framework.Logf("get CPU: [%d, %d]", diff[0], diff[1])
+		if diff[0] > 5e8 && diff[1] < 5e6 {
```

Comment:
how about adding a comment about why the value is `5e8` and `5e6`?

Reply:
Fixed